any-sketch/src/main/cc/any_sketch/crypto/sketch_encrypter.cc

```diff
@@ -187,6 +204,46 @@ absl::StatusOr<std::string> SketchEncrypterImpl::Encrypt(
   return encrypted_sketch;
 }
 
+absl::Status SketchEncrypterImpl::AppendNoiseRegisters(
+    const EncryptSketchRequest::PublisherNoiseParameter&
+        publisher_noise_parameter,
+    int value_count, std::string& encrypted_sketch) {
+  // Lock the mutex since most of the crypto computations here are NOT
+  // thread-safe.
+  absl::WriterMutexLock l(&mutex_);
+  DifferentialPrivacyParams params;
+  params.set_epsilon(publisher_noise_parameter.epsilon());
+  params.set_delta(publisher_noise_parameter.delta());
+  ASSIGN_OR_RETURN(
+      int64_t noise_count,
+      GetTruncatedDiscreteLaplaceDistributedRandomNumber(
+          GetPublisherNoiseOptions(
+              params, publisher_noise_parameter.publisher_count())));
+
+  ASSIGN_OR_RETURN(std::string publisher_noise_register_id_ec,
+                   MapToCurve(kPublisherNoiseRegisterId));
+
+  int64_t noise_register_btyes =
```

Comment:
Can you add checks that this is positive? Also, probably want value_count to be non-negative.

Reply:
done